commercetools/service_custom_applications.go

```diff
@@ -0,0 +1,150 @@
+	Description string                      `json:"description"`
+	URL         string                      `json:"url"`
+	NavbarMenu  CustomApplicationNavbarMenu `json:"navbarMenu"`
+	OAuthScopes []string                    `json:"oAuthScopes,omitempty"`
+}
+
+// CustomApplication describes the structure of a Custom Application stored object.
+type CustomApplication struct {
+	ID          string                      `json:"id"`
+	CreatedAt   string                      `json:"createdAt"`
+	UpdatedAt   string                      `json:"updatedAt"`
+	IsActive    bool                        `json:"isActive"`
+	Name        string                      `json:"name"`
+	Description *string                     `json:"description,omitempty"`
+	URL         string                      `json:"url"`
+	NavbarMenu  CustomApplicationNavbarMenu `json:"navbarMenu"`
+}
+
+// ProjectExtension describes the structure of a project extension stored object.
+type ProjectExtension struct {
+	ID           string              `json:"id"`
+	Applications []CustomApplication `json:"applications"`
+}
+
+// GraphQLResponseProjectExtension describes the structure of the query result for fetching a Custom Application.
+type graphQLResponseProjectExtension struct {
+	ProjectExtension *ProjectExtension `json:"projectExtension"`
+}
+
+// GraphQLResponseProjectExtensionCreation describes the structure of the query result for creating a Custom Application.
+type graphQLResponseProjectExtensionCreation struct {
+	CreateProjectExtensionApplication *ProjectExtension `json:"createProjectExtensionApplication"`
+}
+
+// GraphQLResponseProjectExtensionUpdate describes the structure of the query result for updating a Custom Application.
+type graphQLResponseProjectExtensionUpdate struct {
+	UpdateProjectExtensionApplication     *ProjectExtension `json:"updateProjectExtensionApplication"`
+	ActivateProjectExtensionApplication   *ProjectExtension `json:"activateProjectExtensionApplication"`
+	DeactivateProjectExtensionApplication *ProjectExtension `json:"deactivateProjectExtensionApplication"`
+}
+
+// GraphQLResponseProjectExtensionDeletion describes the structure of the query result for deleting a Custom Application.
+type graphQLResponseProjectExtensionDeletion struct {
+	DeleteProjectExtensionApplication *ProjectExtension `json:"deleteProjectExtensionApplication"`
+}
+
+func (client *Client) CustomApplicationCreate(ctx context.Context, draft *CustomApplicationDraft) (*CustomApplication, error) {
+	var result graphQLResponseProjectExtensionCreation
+
+	query := client.NewGraphQLQuery(`
+		mutation CreateCustomApplicationMutation($draft: ApplicationExtensionDataInput!) {
+			createProjectExtensionApplication(data: $draft) {
+				id
+				applications {
+					id
+				}
+			}
+		}`)
+
+	query.Bind("draft", draft)
+	err := query.Execute(&result, query.ForMerchantCenter())
+	if err != nil {
+		return nil, err
+	}
+	// if len(result.CreateProjectExtensionApplication.Applications) < 1 {
```

Comment:
@emmenko I'm looking at how we can get the created custom app back via graphql. Any ideas? I thought as a workaround to sort the applications object based on last created (although that is also not really safe)